dashboard/app/models/script.rb

```diff
@@ -605,9 +605,13 @@ def csf_finish_url
     end
   end
 
-  def summarize
+  def summarize(user)
```

Comment:
Ideally the user-specific data would be returned by [`UsersHelper#summarize_user_progress`](https://github.com/code-dot-org/code-dot-org/blob/bf839e4235f6357b34fd01d0f2b2593deee65381/dashboard/app/helpers/users_helper.rb#L8).  We've intentionally tried to keep these APIs anonymous & cacheable.

(Same for `ApiController#script_structure` above.)